components/cdc/src/channel.rs

```diff
@@ -245,8 +255,8 @@ impl Sink {
     pub fn unbounded_send(&self, event: CdcEvent, force: bool) -> Result<(), SendError> {
         // Try it's best to send error events.
         let bytes = if !force { event.size() as usize } else { 0 };
-        if bytes != 0 && !self.memory_quota.alloc(bytes) {
-            return Err(SendError::Congested);
+        if bytes != 0 {
+            self.memory_quota.alloc(bytes)?;
```

Comment:
```suggestion
            // Use alloc_high_water to avoid disrupted resolvers, because they
            // share the same memory quota.
            self.memory_quota.alloc_high_water(bytes)?;
```